### ISSUE TITLE
bug(kubernetes): inconsistent scan results for specific directory vs. entire project

### ISSUE THREAD

**Issue by haerter-tss** [Checkmarx/kics#7304]
### Expected Behavior
I have two folders that each contain a values.yaml and a Chart.yaml:

```
Project1/ui/values.yaml
Project1/ui/Chart.yaml
Project1/shared/values.yaml
Project1/shared/Chart.yaml
```

There is a high finding (CWE 798 Use of Hard-coded Credentials) in `ui/values.yaml`, this should be found when scanning the complete projectfolder `Project1` as well as when only scanning `Project1/ui`.
When scanning everything I get a High finding in ui/values.yaml but when I only scan the ui folder the finding is missing.

### Actual Behavior

When scanning the entire project directory the high severity finding is detected. However, when scanning the "ui" directory alone, the high severity finding in `values.yaml` is not detected

(Formatted logs and samples helps us to better understand the issue)

See attached Project1-results.json and ui-results.json. I also attached the sourcecode as ZIP file.

### Steps to Reproduce the Problem

1.  Scan UI folder
`docker run -t -v "/home/user/Project1/ui":/path checkmarx/kics scan -p /path -o "/path/" --report-formats "json" --ci --exclude-categories "Best practices" --disable-full-descriptions`
2.  Scan the entire project
``docker run -t -v "/home/user/Project1":/path checkmarx/kics scan -p /path -o "/path/" --report-formats "json" --ci --exclude-categories "Best practices" --disable-full-descriptions``
3. Compare the results of both scans

### Specifications
-   Version: Keeping Infrastructure as Code Secure v2.1.3
-   Platform: Linux
-   Subsystem: Ubuntu 24.04 / Debian 12.7

[code.zip](https://github.com/user-attachments/files/18058009/code.zip)
[Project1-results.json](https://github.com/user-attachments/files/18058013/Project1-results.json)
[ui-results.json](https://github.com/user-attachments/files/18058014/ui-results.json)

**Comment by haerter-tss** [comment]
Hello! Any news regarding this issue?

**Comment by haerter-tss** [comment]
Hi team, any updates on this issue? Appreciate your help!

**Comment by haerter-tss** [comment]
Hi team, is there anything I can add to help solving this issue?

**Comment by haerter-tss** [comment]
Hello team, I wanted to ask again if anyone had the time to look at this issue?

**Comment by haerter-tss** [comment]
Any news on this topic?

**Comment by AndrePereira123** [comment]
Hello, I was trying to figure out what was happening here, and although I could not find a solution, I found some interesting details.

When running with the log level set to TRACE, I found many files supposedly "ignored":

<img width="639" height="289" alt="Image" src="https://github.com/user-attachments/assets/08a3a7db-e3fa-49ed-a529-c398dd6b443b" />

Also, when looking at the generated payloads, I found that scanning either the "shared" or the "ui" folder by themselves resulted in an empty payload (explains FalseNegative). However, scanning their parent folder does not result in a complete payload either. The payload that results from that only takes the files inside "ui" into consideration.

I tried adding a new "filler" folder. Inside it, I copied the files from the "shared" folder and altered some fields to distinguish them. Now, both the files inside "shared" and "ui" get processed and included in the payload, and the "**password: password**" line in "shared\values.yaml" gets flagged:

<img width="936" height="444" alt="Image" src="https://github.com/user-attachments/assets/58159d7c-e16e-4ca5-8a35-1ba12689bbc4" />

Finally, if I change the filler folder name to something like "z-filler", the folders that get processed properly are "ui" and "z-filler". It seems the engine is not processing the first folder (alphabetically) of the project, and scanning any single folder results in an empty payload.

Scanning files individually works as expected. From what I can tell, the engine itself is not properly handling the scanned files. For some reason, they are chosen to be "ignored", but seemingly are not, as long as they are not located within the first folder of the project.

Given this behavior of only ignoring files in the first folder, this does not seem intended. Moreover, these files seem perfectly acceptable to be scanned, being "ignored" does not seem intended to me but i must say i dont know exactly what those mean in the context of TRACE level logs.

To summarize:
 - Individual files flag as expected;
 - Folders ignore files and generate an empy payload;
 - Folders with folders within ignore the first folder and build the payload with files from the rest.
